src/main/java/graphql/execution/Async.java

```diff
@@ -56,7 +57,16 @@ public void add(CompletableFuture<T> completableFuture) {
         @Override
         public CompletableFuture<List<T>> await() {
             Assert.assertTrue(ix == 0, () -> "expected size was " + 0 + " got " + ix);
-            return CompletableFuture.completedFuture(Collections.emptyList());
+            return typedEmpty();
+        }
+
+
+        // implementation details: infer the type of Completable<List<T>> from a singleton empty
+        private static final CompletableFuture<List<?>> EMPTY = CompletableFuture.completedFuture(Collections.emptyList());
+
+        @SuppressWarnings("unchecked")
+        private static <T> CompletableFuture<T> typedEmpty() {
+            return (CompletableFuture<T>) EMPTY;
         }
```

Comment:
This is a pretty good pattern for CFs that are really a static value - eg one that is completed and the same value